### ISSUE TITLE
Checker hangs during compilation

### ISSUE THREAD

**Issue by jacek-lewandowski** [typetools/checker-framework#6028]
I'm trying to incorporate Checkerframework into the Cassandra project and have a problem as the checker hangs during compilation. After outputting some errors from the analysis, it hung. I can see some significant CPU usage, but there has yet to be progress.

Can I debug the problem - make it print information about the files being processed or investigate it somehow to provide meaningful input in this ticket?

I can also provide a link to the Cassandra feature branch, which needs help with this problem.

**Comment by kelloggm** [comment]
This may or may not be helpful, so take it with a grain of salt. I tried to set up the Checker Framework to run on Cassandra several years ago, in 2019. As I recall, one of the first issues I had was that the framework used more memory than was available by default in the JVM - I think I needed to increase it to something like 24GB of memory before I started to see results. I abandoned the attempt for other reasons (I wasn't a Cassandra developer), though, so I can't say for sure if the same issue is what you're running into now.

I found my [old branch](https://github.com/apache/cassandra/compare/trunk...kelloggm:cassandra:compliance-typecheckers) with the changes I'd made to get these checkers running on Cassandra back then, if that would be helpful to you.

**Comment by msridhar** [comment]
@jacek-lewandowski a link to the branch would be helpful.  Which checkers you are trying to run may also be relevant.

**Comment by jacek-lewandowski** [comment]
Thanks guys, I've figured out that - indeed, it was OOM thrown after quite a long time. Now I have it working. The problem was that it tried to analyse the sources generated by ANTLR, which is extremely long and complex. I simply excluded the generated sources. 

This is the branch I'm working on
https://github.com/jacek-lewandowski/cassandra/tree/ecj-testing

I think this ticket can be closed now
